src/sql/src/plan/scope.rs

```diff
@@ -302,15 +305,24 @@ impl Scope {
             })
     }
 
-    pub fn product(self, right: Self) -> Self {
-        Scope {
+    pub fn product(self, right: Self) -> Result<Self, anyhow::Error> {
+        let l_tables = self.table_names();
+        let r_tables = right.table_names();
+        for l in l_tables {
```

Comment:
Is this better than `l_tables.intersection(r_tables)`?

Reply:
Ok. The other annoyance with this for loop is that if there are multiple conflicting names, it's not deterministic which one will print. Not a huge deal, but would be annoying if any tests look for that. Still fine moving forward as is though.

Reply:
Very fair point; if I dip back into refactoring this for Phillip's comments will make the output deterministic.